pkg/trace/writer/sender.go

```diff
@@ -265,11 +258,18 @@ func (s *sender) sendPayload(p *payload) {
 			// e.g. attempts 4, 8, 16, etc.
 			log.Warnf("Retried payload %d times: %s", r, err.Error())
 		}
+		if p.retries.Load() >= 4 {
```

Comment:
Where did this `4` come from?

Reply:
This is a magic number I picked.

It seems like 4 failed attemts is enough to say that the payload is not going to succeed. However definitely open to discussing what this number *should* be.

I'm not sure how this could/should be parameterized. For instance if the trace agent isn't under any load, we could hang onto payloads indefinitely. However, I'm not sure how valuable that work is relative to the complexity involved. Given the trace agent is usually not just sitting there doing nothing *while also* having payloads waiting to go out , my guess is it's not very valuable to try and adjust the retry count based on internal state.

Happy to hear thoughts.